compiler/test/dotty/tools/dotc/reporting/ErrorMessagesTests.scala

```diff
@@ -304,5 +304,29 @@ class ErrorMessagesTests extends ErrorMessagesTest {
       assertEquals("x", tree.name.show)
     }
 
+  @Test def superQualMustBeParent =
+    checkMessagesAfter("frontend") {
+      """
+        |class A {
+        |  def foo(): Unit = ()
+        |}
+        |
+        |class B {
+        |}
+        |
+        |class C extends A {
+        |  super[B].foo
+        |}
+      """.stripMargin
+    }
+      .expect { (ictx, messages) =>
+        implicit val ctx: Context = ictx
+        val defn = ictx.definitions
 
+        assertMessageCount(1, messages)
+        val SuperQualMustBeParent(qual, cls) :: Nil = messages
+
+        assert(qual.name.toString == "B", s"qualifier was: ${qual.name}")
+        assert(cls.denot.name.toString == "C", s"class was: ${cls.denot.name.toString}")
```

Comment:
You'd have to synthesize or request it from the symbol table anyway. I wouldn't worry about it too much. That being said, as aforementioned, using `.show` is preferable here:

```scala
assertEquals("class B", qual.show)
assertEquals("class C", cls.show)
```

Going down a bit further into the rabbit hole - if you feel that doing string comparisons is watered down: you could also request the symbols from the context by doing a lookup à la:

```scala
val classFromCtx = ctx.getClassIfDefined("full.path.to.ClassName".toTypeName)
```

The problem with doing this is that you're depending on the semantics of `getClassIfDefined` to always give you what you want. So first you'd have to do `assert(classFromCtx.exists, "couldn't get class from context")`. Then you'd have to be sure that the class you got from the context actually is the `class C` that you've defined, so you'd have to test that too. Then you'd do your comparison. For simplicity I think it's easier just to compare the shown strings :)